packages/server/src/fhir/operations/deploy.test.ts

```diff
@@ -22,6 +42,128 @@ describe('Deploy', () => {
     await shutdownApp();
   });
 
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  test('Deploy bot with executableCode attached', async () => {
+    const code = `
+    export async function handler() {
+      console.log('input', input);
+      return input;
+    }
+    `;
+
+    const mockBinaryStorage = new MockBinaryStorage();
+    jest
+      .spyOn(storage, 'getBinaryStorage')
+      .mockImplementation(() => mockBinaryStorage as unknown as storage.BinaryStorage);
+    const binaryStorage = storage.getBinaryStorage();
+    const readBinarySpy = jest.spyOn(binaryStorage, 'readBinary');
+
+    const readStreamToStringSpy = jest
+      .spyOn(streamUtils, 'readStreamToString')
+      .mockImplementation(() => Promise.resolve(code));
+
+    const deployLambdaSpy = jest.spyOn(awsDeploy, 'deployLambda').mockImplementation();
+
+    // Create Binary to serve as storage for code attachment
+    const res1 = await request(app)
+      .post(`/fhir/R4/Binary`)
+      .set('Content-Type', ContentType.FHIR_JSON)
+      .set('Authorization', 'Bearer ' + accessToken)
+      .send({
+        resourceType: 'Binary',
+        contentType: ContentType.JAVASCRIPT,
+      } satisfies Binary);
+    expect(res1.status).toBe(201);
+
+    const binary = res1.body as Binary;
+
+    // Step 2: Create a bot
+    const res2 = await request(app)
+      .post(`/fhir/R4/Bot`)
+      .set('Content-Type', ContentType.FHIR_JSON)
+      .set('Authorization', 'Bearer ' + accessToken)
+      .send({
+        resourceType: 'Bot',
+        name: 'Test Bot',
+        runtimeVersion: 'awslambda',
+        executableCode: { url: `Binary/${binary.id}` },
+      } satisfies Bot);
+    expect(res2.status).toBe(201);
+
+    const bot = res2.body as Bot;
+
+    // Step 3: Deploy the bot
+    const res3 = await request(app)
+      .post(`/fhir/R4/Bot/${bot.id}/$deploy`)
+      .set('Content-Type', ContentType.FHIR_JSON)
+      .set('Authorization', 'Bearer ' + accessToken)
+      .send();
+    expect(res3.status).toBe(200);
+
+    expect(readBinarySpy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        resourceType: 'Binary',
+        contentType: ContentType.JAVASCRIPT,
+      } as Binary)
+    );
+    expect(readStreamToStringSpy).toHaveBeenCalledWith(expect.any(Object));
+    expect(deployLambdaSpy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        ...bot,
+        executableCode: expect.objectContaining({ url: expect.any(String) }),
+        meta: expect.objectContaining({ ...bot.meta, lastUpdated: expect.any(String), versionId: expect.any(String) }),
+      }),
+      code
+    );
+  });
+
+  test('Deprecated: Deploy bot with code parameter', async () => {
```

Comment:
Is this actually deprecated?  I thought we were just making it optional?

Reply:
Good question, I was just kinda going off what Rahul mentioned about deprecating it but I guess that discussion was never conclusive per se.

Philosophically I'm more for deprecating it (reducing amount of ways to do one thing, there's no good reason to keep both ways forever I think), mostly just to homogenize how code is attached to a Bot at some point in the future. But I'm definitely open to be persuaded for it not to be deprecated.

cc: @rahul1